src/components/settings/Settings.cpp

```diff
@@ -34,6 +34,9 @@ void Settings::LoadSettingsFromFile() {
   if (bufferSettings.version == settingsVersion) {
     settings = bufferSettings;
   }
+  if (settings.dfuAndFsMode == DfuAndFsMode::EnabledTillReboot) {
```

Comment:
Although it doesn't make too much of a difference, maybe apply this to `bufferSettings` inside the if clause. That way it's clear that work is being done only when settings are successfully loaded, and it means that settings only gets updated once

Reply:
@mark9064 I would prefer to leave this as it is now. Currently it will fail-secure (set `settings.dfuAndFsMode` to `DfuAndFsMode::Disabled`) if something very unexpected happens. 

@NeroBurner This is the reason why `DfuAndFsMode::EnabledTillReboot` should never get restored from the saved settings on reboot.

Reply:
Sorry if it was unclear, I'm suggesting you do this
```cpp
  if (bufferSettings.version == settingsVersion) {
    if (bufferSettings.dfuAndFsMode == DfuAndFsMode::EnabledTillReboot) {
      bufferSettings.dfuAndFsMode = DfuAndFsMode::Disabled;
    }
    settings = bufferSettings;
  }
```
I believe this does not change how the code behaves at all?

Reply:
This is exactly the thing I would like to keep as is (unless requested by the maintainers to change it).
If possible I would like to keep this code as it is and keep it at the very end of `Settings::LoadSettingsFromFile()`.

The change you suggest would not change how the code behaves.
However in case of refactoring it would increase the chances of the `if (bufferSettings.dfuAndFsMode == DfuAndFsMode::EnabledTillReboot) {` to be forgotten or moved to a code path that is not always executed.

Maybe I should add a code comment above `if (settings.dfuAndFsMode == DfuAndFsMode::EnabledTillReboot) {` telling that this (when needed) switches off DFU and FS access after reboot? 
Or, should I make the requested code change + a code comment?

Reply:
I think I'd personally prefer the suggested change with a comment explaining that this is needed for firmware+files security to work properly.

Maybe it'd be better to fix this in the data format itself thought. What about introducing a new boolean member of Settings.h that's not inside the Settings struct that stores whether DFU+FS is temporarily enabled.
Then `GetDfuAndFsMode` can compute the correct enum value using the setting+the member, and `SetDfuAndFsMode` can unpack the state into Disabled,Disabled+member set,Enabled. That way it will be impossible to break when refactoring and we don't need to mess with saving/loading settings

Reply:
@mark9064 Thanks! Two bool variables (one in `settings` and the other one outside) sounds like a really good idea, I have implemented it (see the recent push).